domain/src/main/kotlin/feature/BackupManager.kt

```diff
@@ -0,0 +1,58 @@
+package ltd.evilcorp.domain.feature
+
+import android.content.res.Resources
+import android.net.Uri
+import androidx.core.os.ConfigurationCompat
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import ltd.evilcorp.core.repository.MessageRepository
+import ltd.evilcorp.core.vo.Message
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.collections.List
+
+@Singleton
+class BackupManager @Inject constructor(
+    private val messageRepository: MessageRepository,
+) {
+
+    companion object {
+        private val mapper = jacksonObjectMapper()
+    }
+
+    private fun List<Message>.generateBackupMessages(locationSave: String): BackupMessages {
+        return BackupMessages(
+            version = 1, // TODO @Akito: Increment version programmatically on major changes.
+            timestamp = SimpleDateFormat(
+                """yyyy-MM-dd'T'HH-mm-ss""",
+                ConfigurationCompat
+                    .getLocales(Resources.getSystem().configuration).get(0),
+            ).format(Date()),
+            locationSaved = locationSave,
+            entries = this,
+        )
+    }
+
+    private fun List<Message>.generateBackupMessagesJString(locationSave: String): String =
+        mapper.writerWithDefaultPrettyPrinter().writeValueAsString(this.generateBackupMessages(locationSave))
+
+    private fun getMessages(publicKey: String): List<Message> =
+        messageRepository.getStatic(publicKey)
+
+    fun generateBackupMessagesJString(publicKey: String, locationSave: Uri): String = getMessages(publicKey)
+        .generateBackupMessagesJString(
+            locationSave.path
+                ?: throw IllegalStateException(
+                    """[backupHistory] Provided Backup Save Location may not be null!""",
+                ),
+        )
+}
+
+data class BackupMessages(
+    val version: Int, /* Different model versions will require different import methods. */
+    val timestamp: String, /* Date & Time of when backup was saved. */
+    val locationSaved: String, /* Absolute path to directory, where backup was saved to. */
```

Comment:
I don't think we need to include `locationSaved`? The file will be available on disk where the user exported it. Even if we include this, using the new storage access framework, we'll get paths that are only visible from inside aTox during the current session, so they'll have literally 0 meaning to the user.

Reply:
The backups are meant to be imported at some point, which is also the reason why there is some bloat in the messages, as the original messages are used, instead of a more reasonable structure, from a user's point of view.

Though yes, the saved location is basically just a port of an earlier idea I had, which is not really valid here, anymore.